pkg/kubectl/cmd/expose.go

```diff
@@ -133,7 +134,11 @@ func RunExpose(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []str
 	}
 	names := generator.ParamNames()
 	params := kubectl.MakeParams(cmd, names)
-	params["default-name"] = info.Name
+	name := info.Name
+	if len(name) > validation.DNS952LabelMaxLength {
+		name = name[:validation.DNS952LabelMaxLength]
```

Comment:
Hrm - we don't do that in other places, and the hash really isn't useful
for users.  If you run expose on the same service twice, having the second
one fail because "already exists" is what you want, rather than it
succeeding.  That's why we have --name.

On Tue, Sep 29, 2015 at 2:23 PM, David Eads <notifications@github.com>
wrote:

> In pkg/kubectl/cmd/expose.go
> <https://github.com/kubernetes/kubernetes/pull/14728#discussion_r40663984>
> :
>
> > @@ -133,7 +134,11 @@ func RunExpose(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []str
> >  	}
> >  	names := generator.ParamNames()
> >  	params := kubectl.MakeParams(cmd, names)
> > -	params["default-name"] = info.Name
> > +	name := info.Name
> > +	if len(name) > validation.DNS952LabelMaxLength {
> > +		name = name[:validation.DNS952LabelMaxLength]
>
> I would prefer adding a hash instead of simply truncating.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/14728/files#r40663984>.
>